diff --git a/compile_pointnet_tfops.sh b/compile_pointnet_tfops.sh
index f8d6d25..b376d70 100644
--- a/compile_pointnet_tfops.sh
+++ b/compile_pointnet_tfops.sh
@@ -4,10 +4,10 @@ TF_CFLAGS=$(python3 -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get
 TF_LFLAGS=$(python3 -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))')
 cd pointnet2/tf_ops/sampling
 
-nvcc -std=c++11 -c -o tf_sampling_g.cu.o tf_sampling_g.cu \
+nvcc -std=c++17 -c -o tf_sampling_g.cu.o tf_sampling_g.cu \
  ${CUDA_INCLUDE} ${TF_CFLAGS} -D GOOGLE_CUDA=1 -x cu -Xcompiler -fPIC
 
-g++ -std=c++11 -shared -o tf_sampling_so.so tf_sampling.cpp \
+g++ -std=c++17 -shared -o tf_sampling_so.so tf_sampling.cpp \
  tf_sampling_g.cu.o ${CUDA_INCLUDE} ${TF_CFLAGS} -fPIC -lcudart ${TF_LFLAGS} ${CUDA_LIB}
 
 echo 'testing sampling'
@@ -15,10 +15,10 @@ python3 tf_sampling.py
  
 cd ../grouping
 
-nvcc -std=c++11 -c -o tf_grouping_g.cu.o tf_grouping_g.cu \
+nvcc -std=c++17 -c -o tf_grouping_g.cu.o tf_grouping_g.cu \
  ${CUDA_INCLUDE} ${TF_CFLAGS} -D GOOGLE_CUDA=1 -x cu -Xcompiler -fPIC
 
-g++ -std=c++11 -shared -o tf_grouping_so.so tf_grouping.cpp \
+g++ -std=c++17 -shared -o tf_grouping_so.so tf_grouping.cpp \
  tf_grouping_g.cu.o ${CUDA_INCLUDE} ${TF_CFLAGS} -fPIC -lcudart ${TF_LFLAGS} ${CUDA_LIB}
 
 echo 'testing grouping'
@@ -26,7 +26,7 @@ python3 tf_grouping_op_test.py
 
  
 cd ../3d_interpolation
-g++ -std=c++11 tf_interpolate.cpp -o tf_interpolate_so.so -shared -fPIC -shared -fPIC ${TF_CFLAGS} ${TF_LFLAGS} -O2
+g++ -std=c++17 tf_interpolate.cpp -o tf_interpolate_so.so -shared -fPIC -shared -fPIC ${TF_CFLAGS} ${TF_LFLAGS} -O2
 echo 'testing interpolate'
 python3 tf_interpolate_op_test.py
 
diff --git a/contact_graspnet/config_utils.py b/contact_graspnet/config_utils.py
index 1815bb5..2e5b7a1 100644
--- a/contact_graspnet/config_utils.py
+++ b/contact_graspnet/config_utils.py
@@ -37,7 +37,7 @@ def load_config(checkpoint_dir, batch_size=None, max_epoch=None, data_path=None,
     config_path = os.path.join(checkpoint_dir, 'config.yaml')
     config_path = config_path if os.path.exists(config_path) else os.path.join(os.path.dirname(__file__),'config.yaml')
     with open(config_path,'r') as f:
-        global_config = yaml.load(f)
+        global_config = yaml.load(f, Loader=yaml.SafeLoader)
 
     for conf in arg_configs:
         k_str, v = conf.split(':')
diff --git a/pointnet2/tf_ops/3d_interpolation/tf_interpolate.cpp b/pointnet2/tf_ops/3d_interpolation/tf_interpolate.cpp
index 0123962..fd38d21 100644
--- a/pointnet2/tf_ops/3d_interpolation/tf_interpolate.cpp
+++ b/pointnet2/tf_ops/3d_interpolation/tf_interpolate.cpp
@@ -17,7 +17,7 @@ REGISTER_OP("ThreeNN")
     .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
         c->set_output(0, c->input(0));
         c->set_output(1, c->input(0));
-        return Status::OK();
+        return OkStatus();
     });
 REGISTER_OP("ThreeInterpolate")
     .Input("points: float32")
@@ -32,7 +32,7 @@ REGISTER_OP("ThreeInterpolate")
         // (b,n,c)
         ::tensorflow::shape_inference::ShapeHandle output = c->MakeShape({c->Dim(dims1, 0), c->Dim(dims2, 1), c->Dim(dims1, 2)});
         c->set_output(0, output);
-        return Status::OK();
+        return OkStatus();
     });
 REGISTER_OP("ThreeInterpolateGrad")
     .Input("points: float32")
@@ -42,7 +42,7 @@ REGISTER_OP("ThreeInterpolateGrad")
     .Output("grad_points: float32")
     .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
         c->set_output(0, c->input(0));
-        return Status::OK();
+        return OkStatus();
     });
 
 float randomf(){
diff --git a/pointnet2/tf_ops/grouping/tf_grouping.cpp b/pointnet2/tf_ops/grouping/tf_grouping.cpp
index c7ca545..36443ca 100644
--- a/pointnet2/tf_ops/grouping/tf_grouping.cpp
+++ b/pointnet2/tf_ops/grouping/tf_grouping.cpp
@@ -26,7 +26,7 @@ REGISTER_OP("QueryBallPoint")
         c->set_output(0, output1);
         ::tensorflow::shape_inference::ShapeHandle output2 = c->MakeShape({c->Dim(dims2, 0), c->Dim(dims2, 1)});
         c->set_output(1, output2);
-        return Status::OK();
+        return OkStatus();
     });
 REGISTER_OP("SelectionSort")
     .Attr("k: int")
@@ -36,7 +36,7 @@ REGISTER_OP("SelectionSort")
     .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
         c->set_output(0, c->input(0));
         c->set_output(1, c->input(0));
-        return Status::OK();
+        return OkStatus();
     });
 REGISTER_OP("GroupPoint")
     .Input("points: float32")
@@ -50,7 +50,7 @@ REGISTER_OP("GroupPoint")
         // batch_size * npoints * nsample * channels
         ::tensorflow::shape_inference::ShapeHandle output = c->MakeShape({c->Dim(dims2, 0), c->Dim(dims2, 1), c->Dim(dims2, 2), c->Dim(dims1, 2)});
         c->set_output(0, output);
-        return Status::OK();
+        return OkStatus();
     });
 REGISTER_OP("GroupPointGrad")
     .Input("points: float32")
@@ -59,7 +59,7 @@ REGISTER_OP("GroupPointGrad")
     .Output("grad_points: float32")
     .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
         c->set_output(0, c->input(0));
-        return Status::OK();
+        return OkStatus();
     });
 
 
diff --git a/pointnet2/tf_ops/sampling/tf_sampling.cpp b/pointnet2/tf_ops/sampling/tf_sampling.cpp
index fb3dd28..efb99fb 100644
--- a/pointnet2/tf_ops/sampling/tf_sampling.cpp
+++ b/pointnet2/tf_ops/sampling/tf_sampling.cpp
@@ -23,7 +23,7 @@ REGISTER_OP("ProbSample")
     // batch_size * npoints
     ::tensorflow::shape_inference::ShapeHandle output = c->MakeShape({c->Dim(dims2, 0), c->Dim(dims2, 1)});
     c->set_output(0, output);
-    return Status::OK();
+    return OkStatus();
   });
 REGISTER_OP("FarthestPointSample")
   .Attr("npoint: int")
@@ -36,7 +36,7 @@ REGISTER_OP("FarthestPointSample")
     TF_RETURN_IF_ERROR(c->GetAttr("npoint", &npoint));
     ::tensorflow::shape_inference::ShapeHandle output = c->MakeShape({c->Dim(dims1, 0), npoint});
     c->set_output(0, output);
-    return Status::OK();
+    return OkStatus();
   });
 REGISTER_OP("GatherPoint")
   .Input("inp: float32")
@@ -50,7 +50,7 @@ REGISTER_OP("GatherPoint")
     // batch_size * npoints * 3
     ::tensorflow::shape_inference::ShapeHandle output = c->MakeShape({c->Dim(dims1, 0), c->Dim(dims2, 1), c->Dim(dims1, 2)});
     c->set_output(0, output);
-    return Status::OK();
+    return OkStatus();
   });
 REGISTER_OP("GatherPointGrad")
   .Input("inp: float32")
@@ -59,7 +59,7 @@ REGISTER_OP("GatherPointGrad")
   .Output("inp_g: float32")
   .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
     c->set_output(0, c->input(0));
-    return Status::OK();
+    return OkStatus();
   });
 
 void probsampleLauncher(int b,int n,int m,const float * inp_p,const float * inp_r,float * temp,int * out);
